pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2686,13 +2652,8 @@ private RetentionPolicies getRetentionPolicies(TopicName topicName, TopicPolicie
         return retentionPolicies;
     }
 
-    protected void internalRemoveBacklogQuota(AsyncResponse asyncResponse,
-            BacklogQuota.BacklogQuotaType backlogQuotaType) {
-        internalSetBacklogQuota(asyncResponse, backlogQuotaType, null);
-    }
-
-    protected void internalGetRetention(AsyncResponse asyncResponse, boolean applied){
-        RetentionPolicies retentionPolicies = getTopicPolicies(topicName)
+    protected RetentionPolicies internalGetRetention(boolean applied) {
```

Comment:
Agree with you, I noticed the current implementation is not in a consistent way, I have a local change for handling this case, but don't want to use this PR because not easy to review. So for this PR I don't want to change any existing internal implementation of the topic policy, just make sure we can avoid the redirect loop.